gobblin-core/src/test/java/gobblin/filesystem/InstrumentedHDFSFileSystemTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.filesystem;
+
+import java.net.URI;
+import java.util.UUID;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class InstrumentedHDFSFileSystemTest {
+
+  /**
+   * This test is disabled because it requires a local hdfs cluster at localhost:8020, which requires installation and setup.
+   * Changes to {@link InstrumentedHDFSFileSystem} should be followed by a manual run of this tests.
+   *
+   * TODO: figure out how to fully automate this test.
```

Comment:
Yes, it will take some effort to automate such tests. It would be nice to have an open Github issue tracking it so that we don't forget about it.